### ISSUE TITLE
VGG case

### ISSUE THREAD

**Issue by dj163110** [bigcatml/myia#5]
use case for VGG net

**Comment by invisiblewei** [comment]
# tasks

: nchw

## layers/ compute unit
- [x] base conv
- [ ] check conv param for vgg 
- [ ] max pooling 
- [x] relu
- [x] dense 
- [ ] softmax loss(corss entropy) for training
- [ ] softmax for testing
## data
- [ ] data loader, shuffle
- [ ] loader cifer by data loader
- [ ] img preprocess, flip, scale
## solver
- [x] sgd
- 
## check point

## network
- [ ] design small vgg, reduce channel num,
- [ ] per train with

**Comment by invisiblewei** [comment]
> also from my point of view the image augmentation part has nothing to do with myia itself so we may borrow something from https://github.com/aleju/imgaug or just the keras one.

but as a compute graph, easy preprocess like minus mean and scale in the same graph, can reduce  total computation.

**Comment by xiaoleitw** [comment]
sorry... were considered as affine + Crop +Noise such as:

    datagen = ImageDataGenerator(
        featurewise_center=False,  # set input mean to 0 over the dataset
        samplewise_center=False,  # set each sample mean to 0
        featurewise_std_normalization=False,  # divide inputs by std of the dataset
        samplewise_std_normalization=False,  # divide each input by its std
        zca_whitening=False,  # apply ZCA whitening
        zca_epsilon=1e-06,  # epsilon for ZCA whitening
        rotation_range=0,  # randomly rotate images in the range (degrees, 0 to 180)
        # randomly shift images horizontally (fraction of total width)
        width_shift_range=0.1,
        # randomly shift images vertically (fraction of total height)
        height_shift_range=0.1,
        shear_range=0.,  # set range for random shear
        zoom_range=0.,  # set range for random zoom
        channel_shift_range=0.,  # set range for random channel shifts
        # set mode for filling points outside the input boundaries
        fill_mode='nearest',
        cval=0.,  # value used for fill_mode = "constant"
        horizontal_flip=True,  # randomly flip images
        vertical_flip=False,  # randomly flip images
        # set rescaling factor (applied before any other transformation)
        rescale=None,
        # set function that will be applied on each input
        preprocessing_function=None,
        # image data format, either "channels_first" or "channels_last"
        data_format=None,
        # fraction of images reserved for validation (strictly between 0 and 1)
        validation_split=0.0)
